syntax = "proto2";

package phxqueue.store.proto;
import "phxqueue/comm/proto/comm.proto";

message SyncCtrlInfo {
    message QueueDetail {
        optional int32 queue_id = 1;
        message ConsumerGroupDetail {
            optional int32 consumer_group_id = 1;
            optional uint64 prev_cursor_id = 2;
        }
        repeated ConsumerGroupDetail consumer_group_details = 2;
    }
    repeated QueueDetail queue_details = 1;
}


message StorePaxosArgs {
    optional comm.proto.AddRequest add_req = 1;
    optional SyncCtrlInfo sync_ctrl_info = 101;
    optional comm.proto.Addr master_addr = 102;
    optional uint64 timestamp = 201;
}

